### PR TITLE
adding lesson07 work

### PR DESCRIPTION
**Pull request by stefanjp1**
Adding lesson07 work

### CODE REVIEW

File: students/stefanjp1/lesson07/html_render.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+
+class Element():
+    tag_name = ''
```
Comment (charlesrobison) [inline students/stefanjp1/lesson07/html_render.py]: Missing 'html' to assign to tag_name

File: students/stefanjp1/lesson07/html_render.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    def render(self, file_out, cur_ind = ""):
+        file_out.write("<!DOCTYPE html>\n")
+        
+        super().render(file_out, cur_ind)
+    
+
+class Body(Element):
+    tag_name = 'body'
+    
+    def __init__(self, content=None, **kwargs):
```
Comment (charlesrobison) [inline students/stefanjp1/lesson07/html_render.py]: __init__ not necessary here as you are inheriting from Element.

File: students/stefanjp1/lesson07/html_render.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        self.content = []
+        if content != None:
+            self.content.append(TextWrapper(content))
+        self.kwargs = kwargs 
+
+
+class Head(Element):
+    tag_name = 'head'
+    
+    def __init__(self, content=None, **kwargs):
```
Comment (charlesrobison) [inline students/stefanjp1/lesson07/html_render.py]: __init__ not necessary here as you are inheriting from Element.

File: students/stefanjp1/lesson07/html_render.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        file_out.write("<{}{}>".format(self.tag_name, tags))
+        for block in self.content:
+            block.render(file_out)
+        file_out.write("</{}>\n".format(self.tag_name))
+
+
+class Title(OneLineTag):
+    tag_name = 'title'
+    
+    def __init__(self, content=None, **kwargs):
```
Comment (charlesrobison) [inline students/stefanjp1/lesson07/html_render.py]: __init__ not necessary here as you are inheriting from OneLineTag.

File: students/stefanjp1/lesson07/html_render.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+class Hr(SelfClosingTag):
+    tag_name = 'hr'
+    
+class Br(SelfClosingTag):
+    tag_name = 'br'
+    
+class A(OneLineTag):
+    tag_name = 'a'
+    
+    def __init__(self, link, content=None, **kwargs):
```
Comment (charlesrobison) [inline students/stefanjp1/lesson07/html_render.py]: __init__ not necessary here as you are inheriting from OneLineTag.

File: students/stefanjp1/lesson07/html_render.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        self.content = []
+        if content != None:
+            self.content.append(TextWrapper(content))
+        self.kwargs = kwargs  
+        self.kwargs['href'] = link
+        
+class Ul(Element):
+    tag_name = 'ul'
+    
+    def __init__(self, content=None, **kwargs):
```
Comment (charlesrobison) [inline students/stefanjp1/lesson07/html_render.py]: __init__ not necessary here as you are inheriting from Element.

**Comment by natasha-aleksandrova** [comment]
please review the comments and we can adjust the grade if you would like to resubmit, thanks!

**Pull request merged**